test/assertions_test.rb

```diff
@@ -87,13 +87,43 @@ def test_assert_statsd_call
       end
     end
 
+    assert_no_assertion_triggered do
+      foo_1_metric_tags = { times: 2, tags: ['foo:1'] }
+      foo_2_metric_tags = { times: 1, tags: ['foo:2'] }
+      @test_case.assert_statsd_increments('counter', [foo_1_metric_tags, foo_2_metric_tags]) do
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 1 })
+        StatsD.increment('counter', tags: { foo: 2 })
+      end
+    end
```

Comment:
I don't think we ever care what order metrics appear in. At least I can't think of any use cases for that. I wouldn't add support for that unless someone explicitly asks for it.